ironfish/src/blockSyncer.ts

```diff
@@ -281,8 +283,14 @@ export class BlockSyncer<
       block.header.sequence <= this.blocksForProcessing[0].block.header.sequence
     ) {
       this.blocksForProcessing.unshift({ block, fromPeer, type })
+      if (this.blocksForProcessing.length > this.blocksForProcessingMax) {
```

Comment:
If we suddenly drop blocks here, what will cause these blocks to get re-requested? This seems like it's going to cause some issues syncing.

Reply:
Suggestions for improving this so the list of blocks to process doesn't grow unbounded? I think the answer to most of these is "Fix bugs in the syncer so it syncs"

Reply:
hmm.. what if we additionally split out the queue to be for gossiped blocks (and we can do that with an lru that we have in other places of the codebase already) and for syncing blocks? if the memory is coming from gossiped blocks, we could put bounds around that specifically

Reply:
Updated to add two queues